File: 2015/day3/first.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>

bool already_visited(std::pair<int, int> pos,
			std::vector<std::pair<int, int>> &positions) {
    for (auto &position : positions) {
	if (position == pos) {
	return true;
	}
    }
    return false;
};

int main() {

  std::ifstream file("./input.txt");

  if (!file.is_open()) {
    std::cerr << "Could not open file\n";
    return -1;
  }

  std::string line{};

  std::vector<std::pair<int, int>> positions;

  positions.push_back({0, 0});
  std::pair<int, int> current_position = {0, 0};
  int sum = 0;

  while (std::getline(file, line)) {
    for (auto &i : line) {
      if (i == '^') {
	current_position.second -= 1;
      }
      else if (i == 'v') {
	current_position.second += 1;
      } else if (i == '>') {
	current_position.first += 1;
      } else if (i == '<') {
	current_position.first -= 1;
      }
      std::cout << current_position.first << ' ' << current_position.second << std::endl;
      if (!already_visited(current_position, positions)) {
	sum += 1;
      }
      positions.push_back(current_position);
    }
  }

  std::cout << sum << std::endl;

  file.close();

  return 0;
}

File: 2017/day4/first.py
if __name__ == "__main__":

    with open("./input.txt", 'r') as file:
        lines = [line for line in file.readlines()]

    sum = 0

    for line in lines:
        seen_words = []

        valid = True

        current_word = ""
        for char in line:
            if char == ' ' or char == '\n':
                if current_word in seen_words:
                    valid = False
                    break
                else:
                    seen_words.append(current_word)

                current_word = ""
            else:
                current_word += char

        if valid:
            sum += 1

    print(sum)
